text/text.go

```diff
@@ -4,123 +4,174 @@ import (
+	if isCustom {
+		var b bytes.Buffer
+		var titleDims []string
+		dt := time.Now()
+		issuedDate := dt.Format("01-02-2006")
 
-	b.WriteString(fmt.Sprintf("Title: %s\n", m.Title))
-	b.WriteString(fmt.Sprintf("Description: %s\n", m.Description))
-	if m.Publisher != nil {
-		b.WriteString(fmt.Sprintf("Publisher: %s\n", *m.Publisher))
-	}
-	b.WriteString(fmt.Sprintf("Issued: %s\n", m.ReleaseDate))
-	b.WriteString(fmt.Sprintf("Next Release: %s\n", m.NextRelease))
-	b.WriteString(fmt.Sprintf("Identifier: %s\n", m.Title))
-	if m.Keywords != nil {
-		b.WriteString(fmt.Sprintf("Keywords: %s\n", *m.Keywords))
-	}
-	b.WriteString(fmt.Sprintf("Language: %s\n", "English"))
-	if m.Contacts != nil {
-		contacts := *m.Contacts
-		if len(contacts) > 0 {
-			b.WriteString(fmt.Sprintf("Contact: %s, %s, %s\n", contacts[0].Name, contacts[0].Email, contacts[0].Telephone))
+		for _, d := range m.Version.Dimensions {
+			titleDims = append(titleDims, d.Label)
 		}
-	}
-	if len(m.Temporal) > 0 {
-		b.WriteString(fmt.Sprintf("Temporal: %s\n", m.Temporal[0].Frequency))
-	}
-	b.WriteString(fmt.Sprintf("Latest Changes: %s\n", m.LatestChanges))
-	b.WriteString(fmt.Sprintf("Periodicity: %s\n", m.ReleaseFrequency))
-	b.WriteString("Distribution:\n")
-	for k, v := range m.Downloads {
-		b.WriteString(fmt.Sprintf("\tExtension: %s\n", k))
-		b.WriteString(fmt.Sprintf("\tSize: %s\n", v.Size))
-		b.WriteString(fmt.Sprintf("\tURL: %s\n\n", v.URL))
-	}
-	b.WriteString(fmt.Sprintf("Unit of measure: %s\n", m.UnitOfMeasure))
-	b.WriteString(fmt.Sprintf("License: %s\n", m.License))
-	if m.Methodologies != nil {
-		b.WriteString(fmt.Sprintf("Methodologies: %s\n", *m.Methodologies))
-	}
-	if m.Publications != nil {
-		b.WriteString(fmt.Sprintf("Publications: %s\n", *m.Publications))
-	}
-	if m.RelatedDatasets != nil {
-		b.WriteString(fmt.Sprintf("Related Links: %s\n", *m.RelatedDatasets))
-	}
-	b.WriteString(fmt.Sprintf("Canonical Topic: %s\n", m.CanonicalTopic))
-	if len(m.Subtopics) > 0 {
-		b.WriteString(fmt.Sprintf("Subtopics: %s\n", m.Subtopics))
-	}
-	b.WriteString(fmt.Sprintf("Survey: %s\n", m.Survey))
-
-	// New fields
-	// Combine alerts and useage notes as in csvw?
-	if m.Alerts != nil && len(*m.Alerts) != 0 {
-		b.WriteString(fmt.Sprintf("Alerts:\n"))
-
-		for _, a := range *m.Alerts {
-			b.WriteString(
-				fmt.Sprintf(
-					"Date: %s\nDescription: %s\nType: %s\n",
-					a.Date,
-					a.Description,
-					a.Type,
-				),
-			)
-		}
-	}
 
-	b.WriteString("Usage Notes:\n")
-	if m.DatasetDetails.UsageNotes != nil {
-		for _, n := range *m.DatasetDetails.UsageNotes {
-			b.WriteString(fmt.Sprintf("Title: %s\nNote: %s\n\n", n.Title, n.Note))
+		b.WriteString(fmt.Sprintf("Title: %s\n", titleDims))
+		b.WriteString(fmt.Sprintf("Issued: %s\n", issuedDate))
+		b.WriteString(fmt.Sprintf("Language: %s\n", "English"))
+		b.WriteString("Distribution:\n")
+		for k, v := range m.Downloads {
```

Comment:
Is it not possible to do the same for loop for non custom dataset? Like so:

```
		for k, v := range m.Downloads {
			b.WriteString(fmt.Sprintf("\tExtension: %s\n", k))
			b.WriteString(fmt.Sprintf("\tSize: %s\n", v.Size))
			b.WriteString(fmt.Sprintf("\tURL: %s\n\n", v.URL))
		}
```

If yes, then can move this to a separate function and can be used by both sections

Reply:
I don't think this is possible as v.URL does not contain the correct url, therefore they need to be processed separately depending on the file type